gandalf.js

```diff
@@ -1,104 +1,105 @@
 /**
```

Comment:
This new feature lacks unit tests. Could you add some to ensure everything works correctly?